### PR TITLE
feat: make buffer size for agent data configurable

### PR DESCRIPTION
**Pull request by kruskall**
Add ELASTIC_APM_LAMBDA_AGENT_DATA_BUFFER_SIZE variable to set
the buffer size.

Closes https://github.com/elastic/apm-aws-lambda/issues/28

### CODE REVIEW

File: apm-lambda-extension/app/app.go [hunk 1]
```diff
@@ -84,6 +84,14 @@ func New(opts ...configOption) (*App, error) {
 		apmOpts = append(apmOpts, apmproxy.WithReceiverAddress(fmt.Sprintf(":%s", port)))
 	}
 
+	if bufferSize := os.Getenv("ELASTIC_APM_LAMBDA_AGENT_DATA_BUFFER_SIZE"); bufferSize != "" {
+		if size, err := strconv.Atoi(bufferSize); err != nil {
+			return nil, err
+		} else {
+			apmOpts = append(apmOpts, apmproxy.WithAgentDataBufferSize(size))
+		}
```
Comment (axw) [inline apm-lambda-extension/app/app.go:92]:
```suggestion
		}
		apmOpts = append(apmOpts, apmproxy.WithAgentDataBufferSize(size))
```

**Review by axw — COMMENTED** [review]
Looks good, just one nit.

I think we should also update the docs: https://github.com/elastic/apm-aws-lambda/blob/main/docs/monitoring-aws-lambda.asciidoc#configuration-options

File: apm-lambda-extension/app/app.go (hunk 1 above)
Comment (kruskall) [reply apm-lambda-extension/app/app.go]: Changed 👍

**Comment by kruskall** [comment]
Updated the documentation and added a changelog entry.

**Review by axw — APPROVED** [review]

**Pull request merged**